### ISSUE TITLE
Test case 5 sequence of bookings

### ISSUE THREAD

**Issue by eesean** [nus-cs2030/2223-s2#282]
Hi all,

I used a comparator to compare between bookings by cost followed by waiting time, my first test case showed as follows:

<img width="431" alt="comparator" src="https://user-images.githubusercontent.com/110977266/221362921-0e82f0e0-40c6-4f66-a7fe-a7d7b39f6c0d.png">

Found it strange that the sequence for the second test case is correct. If I have another class extending from Booking intended to show the alternative booking, should the class use the same compareTo method that I have implemented in my Booking class?

**Comment by haqanh** [comment]
Is it possible that your comparator return value is not correct? (i.e. 1/-1 is not correct in the direction of comparison that you want)

If that doesn't work, for me what I did was I implement a new comparator with a different comapreTo method (as now you compare Price then WaitTime, but keep both possible booking instead of just the best one). 

Hope this helps!

**Comment by eesean** [comment]
Hi just went to check, apparently there was an issue with the other class inheriting from Booking. Because it extends from Booking, the comparison is still by the cheapest possible service the driver can offer and not the fare for the booking, hence I created a method for both classes to determine the fares for each possible booking associated with a driver. Thanks for the tip tho!

**Comment by haoyanggggg** [comment]
hi can u elaborate on the solution bc im also facing this issue and have no idea how to fix it! also got the same exact output and used another class inheriting from booking too. however my booking comparator is not related to compareTo methods in both booking classes hence i hv no clue how to fix it

**Comment by eesean** [comment]
In my case when I created another class (say class A) directly inheriting from Booking, the compareTo method in my booking class compares based on the smallest fare that the driver can offer (meaning class A uses this compareTo method even though it is supposed to show the more expensive fare). Although the toString( ) output is correct, the ordering becomes wrong as the comparison done is actually between the two cheapest services - meaning $3.33 SMA7890 and $4.40 SHA1234. What I meant by toString output is correct is that the other booking class has its own toString method hence the correct behaviour is evoked (i.e. $.3.33 and $4.40 for SMA7890, $.4.40 and $8.60 for SHA1234).

To resolve this, I created a general method (e.g. getFare) in each class, but for the respective classes I compute the cheapest and most expensive fare for the same driver. I use getFare in my compareTo to compare one booking with another, regardless of whether it is the cheapest fare the driver can offer. Hope this makes sense!
